src/Database/PostgreSQL/PQTypes/Model/Migration.hs

```diff
@@ -57,6 +59,17 @@ data MigrationAction m =
       TableIndex
 #endif
 
+  -- | Migration for modifying columns. Parameters are:
+  --
+  -- SQL that will be used for the cursor.
+  --
+  -- Function that takes a list of primary keys provided by the cursor SQL and
+  -- runs an arbitrary computation within MonadDB. The function might be called
+  -- repeatedly depending on the number of primary keys. See the last argument.
+  --
+  -- Number of primary keys fetched at once by the cursor SQL.
+  | forall t . FromSQL t => ModifyColumnMigration SQL ([t] -> m ()) Int
```

Comment:
This should be `FromRow` as there might be multi-column primary keys.

Reply:
Are you sure? From looking at https://hackage.haskell.org/package/hpqtypes-1.9.1.0/docs/Database-PostgreSQL-PQTypes-FromRow.html#t:FromRow, FromSQL seems to be a "superset" of FromRow anyway. Also, FromRow doesn't seem to have instances for basic types.

Reply:
Never mind, I understand why it's needed now. I've updated the PR.